### PR TITLE
Python 3.13 support

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Python 3.13 support changes:
- Update doc string modification in n_jobs support utilities
- Change Python3.12_Sklearn1.5 CondaEnv CI job to Python3.13_Sklearn1.5 

Applied as patch in conda-forge: https://github.com/conda-forge/scikit-learn-intelex-feedstock/pull/49

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

N/A

### CODE REVIEW

File: daal4py/sklearn/_n_jobs_support.py [hunk 1]
```diff
@@ -212,14 +213,16 @@ def init_with_n_jobs(self, *args, n_jobs=None, **kwargs):
...
         ):
-            parameters_doc_tail = "\n    Attributes"
-            n_jobs_doc = """
-    n_jobs : int, default=None
-        The number of jobs to use in parallel for the computation.
-        ``None`` means using all physical cores
-        unless in a :obj:`joblib.parallel_backend` context.
-        ``-1`` means using all logical cores.
-        See :term:`Glossary <n_jobs>` for more details.
+            # Python 3.13 removed extra tab in class doc string
```
Comment (david-cortes-intel) [inline daal4py/sklearn/_n_jobs_support.py:216]: Curious about this. The docstring is meant to be interpreted by the Sphinx plugin "napoleon", which is a python package. Should thus be independent of python version. Do you have a screenshot of how it ends up looking like if you leave the tab as it is?

Comment (Alexsandruss) [reply daal4py/sklearn/_n_jobs_support.py:216]: - `n_jobs` was not added to applicable classes because matching substring of attributes tail was not found
- There is a sklearnex test for n_jobs in docstrings, so it failed initially
- There are many other renderers of docstring like VSCode and other IDEs

Comment (david-cortes-intel) [reply daal4py/sklearn/_n_jobs_support.py:216]: Thanks. Can confirm that the issue with the docstrings happens with python3.13, and that this PR solves it.

**Review by david-cortes-intel — APPROVED** [review]
